handlers.go

```diff
@@ -426,25 +426,34 @@ func (s *Server) handleAddIpfs(c echo.Context, u *User) error {
 		return err
 	}
 
+	filename := params.Name
+	if filename == "" {
+		filename = params.Root
+	}
+
 	var cols []*CollectionRef
 	if params.Collection != "" {
 		var srchCol Collection
 		if err := s.DB.First(&srchCol, "uuid = ? and user_id = ?", params.Collection, u.ID).Error; err != nil {
 			return err
 		}
 
-		var colp *string
-		if params.CollectionPath != nil {
+		// if collectionPath is "" or nil, put the file on the root dir
+		defaultPath := "/"
+		colp := &defaultPath
+		if params.CollectionPath != nil && *params.CollectionPath != "" {
 			p, err := sanitizePath(*params.CollectionPath)
 			if err != nil {
 				return err
 			}
 			colp = &p
 		}
+		// TODO: path is a dir or is the entire path with file? (if we put / the file is named . when listing)
+		pathWithFilename := *colp + filename
```

Comment:
FWIW, the path as it's stored in the database is the absolute path, including filename :)

Reply:
Are you sure? I remember one of the issues was exactly when you submitted an empty (`""`, not `nil`) filename when uploading a file.